buildstock_fetch/ev_demand.py

```diff
@@ -106,89 +145,108 @@ def __init__(self, metadata_path: Optional[str] = None):
             -2.0659e-10,  # a_5 (quintic term)
         ])
 
-    def load_metadata(self, metadata_path: Optional[str] = None) -> pl.DataFrame:
+    def fit_vehicle_ownership_model(self, pums_df: pl.DataFrame) -> Any:
         """
-        Load and parse the ResStock metadata parquet file.
+        Fit a multinomial logistic regression model to predict number of vehicles per household using PUMS data.
+        Results limited to 0, 1, or 2 vehicles.
 
         Args:
-            metadata_path: Path to the metadata parquet file. If None, uses self.metadata_path
+            pums_df: DataFrame with PUMS household data, including 'occupants', 'income', 'metro', 'vehicles'
 
         Returns:
-            DataFrame with columns including 'building_id', 'occupants', 'inc', 'metro_area' (PUMA), etc.
+            Trained model object
         """
-        path = metadata_path or self.metadata_path
-        if path is None:
-            raise MetadataPathError()
+        # Preprocess data: replace vehicles > 2 with 2, drop nulls, encode categorical
+        pums_df = pums_df.with_columns(
```

Comment:
I'd move feature engineering to a separate method, so that this one just takes whatever features we've got and does the training, and the feature engineering method can evolve as we add features.